### DISCUSSION TITLE
Why is `String.EndsWith()` on `.NET 8` two times slower than on `.NET Framework 4.8`, in Chinese and Japanese `CultureInfo`?

### DISCUSSION THREAD

**Discussion by JasonWei512** [dotnet/runtime#97200] [Q&A] [answered]
## Full test project and result

[StringEndsWithBenchmark.zip](https://github.com/dotnet/runtime/files/13985807/StringEndsWithBenchmark.zip)

## Example

On `.NET 8` ...
```csharp
"Hello World".EndsWith("World");
```
... takes 278% time in Chinese `CultureInfo` compared to `.NET Framework 4.8`. 

|Runtime|CultureInfo|Mean|Error|StdDev|Ratio|RatioSD|
|--- |--- |--- |--- |--- |--- |--- |
|.NET 8.0|zh-CN|729.141 ns|5.5568 ns|4.9259 ns|2.78|0.03|
|.NET Framework 4.8|zh-CN|261.966 ns|2.8408 ns|2.6573 ns|1.00|0.00|
|.NET 8.0|ja-JP|684.399 ns|7.3567 ns|6.8815 ns|4.10|0.06|
|.NET Framework 4.8|ja-JP|166.677 ns|2.5655 ns|2.2742 ns|1.00|0.00|
|.NET 8.0|en-US|8.513 ns|0.0718 ns|0.0672 ns|0.13|0.00|
|.NET Framework 4.8|en-US|63.882 ns|0.4042 ns|0.3781 ns|1.00|0.00|
|.NET 8.0|InvariantCulture|8.748 ns|0.1012 ns|0.0898 ns|0.15|0.00|
|.NET Framework 4.8|InvariantCulture|58.908 ns|0.7907 ns|0.7009 ns|1.00|0.00|

## Environment

```
BenchmarkDotNet v0.13.12, Windows 11 (10.0.22631.3007/23H2/2023Update/SunValley3)
AMD Ryzen 7 7840S with Radeon 780M Graphics, 1 CPU, 16 logical and 8 physical cores
  [Host]             : .NET Framework 4.8.1 (4.8.9181.0), X64 RyuJIT VectorSize=256
  .NET 8.0           : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
  .NET Framework 4.8 : .NET Framework 4.8.1 (4.8.9181.0), X64 RyuJIT VectorSize=256
```

**Comment by huoyaoyuan** [comment, marked as answer]
The default globalization implementation is ICU on .NET Core. See https://learn.microsoft.com/dotnet/core/extensions/globalization-icu . You can compare the performance of NLS mode mentioned in the doc.

**Comment by hez2010** [comment]
You may want to compare string with `Ordinal` culture to treat string as binary sequence:
```cs
"Hello World".EndsWith("World", StringComparison.Ordinal);
```
